### DISCUSSION TITLE
docker build error

### DISCUSSION THREAD

**Discussion by gaecoli** [getredash/redash#7143] [Q&A]
RUN yarn install:
0.276 yarn install v1.22.22
0.307 info No lockfile found.
0.308 $ cd viz-lib && yarn link --link-folder ../.yarn
0.498 yarn link v1.22.22
0.538 success Registered "@redash/viz".
0.538 info You can now run `yarn link "@redash/viz"` in the projects where you want to use this package and it will be used instead.
0.538 Done in 0.04s.
0.548 [1/5] Validating package.json...
0.552 [2/5] Resolving packages...
10.00 warning @redash/viz > plotly.js > color-rgba > color-space > mumath@3.3.4: Redundant dependency in your project.
18.69 error Couldn't find the binary git
18.69 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
------
Dockerfile:34
--------------------
  32 |     # RUN if [ "x$skip_frontend_build" = "x" ] ; then yarn --frozen-lockfile --network-concurrency 1; fi
  33 |     # RUN yarn --frozen-lockfile --network-concurrency 1
  34 | >>> RUN yarn install
  35 |
  36 |     COPY --chown=redash client /frontend/client
--------------------
ERROR: failed to solve: process "/bin/sh -c yarn install" did not complete successfully: exit code: 1

**Comment by justinclift** [comment]
This sounds like the main problem:

`Couldn't find the binary git`

Isn't `git` one of the default programs on macOS?

Note that I'm currently writing this on a Linux box, and won't have access to my macOS dev machine for a few hours yet. :wink:

**Reply by justinclift** [reply]
Oh.  That's the `yarn` inside the docker build process.  Sounds like one of our dependencies might have changed in a way that we need to adjust for.  I'll look in a few hours if no-one else gets to it first. :smile:

**Reply by justinclift** [reply]
@gaecoli Was this when running on your local desktop, or was it one of the CI builds failing, or something else? :smile:

**Reply by gaecoli** [reply]
> @gaecoli Was this when running on your local desktop, or was it one of the CI builds failing, or something else? 😄

Sorry, i build it on ubuntu 20.24 and i used docker build. I found it always failed. Maybe yarn install libs failed or yarn build failed.

**Reply by justinclift** [reply]
No worries. :smile:

Which architecture is that with?  `x86_64` or `ARM64` or something else?

**Reply by gaecoli** [reply]
> No worries. 😄
> 
> Which architecture is that with? `x86_64` or `ARM64` or something else?

Oh! I use `x86_64` for a new server.

**Reply by lucydodo** [reply]
I was not able to reproduce the issue, but I was able to see a similar error when i deleted `git` from my system.
Do you have `git` installed on your system? If not, you can install it with the following command: `sudo apt install git`

**Reply by gaecoli** [reply]
> I was not able to reproduce the issue, but I was able to see a similar error when i deleted `git` from my system. Do you have `git` installed on your system? If not, you can install it with the following command: `sudo apt install git`

Sure. I will try it lately.

**Reply by justinclift** [reply]
> for a new server
Oh.  Is the Redash installation for the new server intended for development, or more for people to use like a real deployment?

If it's for a real deployment, then the setup.sh script in [the setup repo](https://github.com/getredash/setup/) is the better thing to use as that's what it's for. :smile:

**Comment by gaecoli** [comment]
When i used tagv10.x.0 it works, but i found it's too slow.
